src/services/formatting/formatting.ts

```diff
@@ -387,7 +389,10 @@ namespace ts.formatting {
 
             let indentation = inheritedIndentation;
             if (indentation === Constants.Unknown) {
-                if (isSomeBlock(node.kind)) {
+                if (isIndentPreventedChildNode(parent.kind, node.kind)) {

```

Comment:
@mhegazy I currently have no idea so I just inlined it. I agree that the closing node is similar to closing tokens but I think nodes and tokens are being processed with some differences by different functions so a merged helper may not really simplify the code.